inject.js

```diff
@@ -294,6 +294,7 @@ module.exports = function(acorn) {
       node.attributes.push(this.jsx_parseAttribute());
     node.selfClosing = this.eat(tt.slash);
     this.expect(tt.jsxTagEnd);
+    if (!node.name) delete node.name;
```

Comment:
`delete` usually hurts performance quite a bit. Could you please read `jsx_parseElementName` into a variable instead and assign to the property only if it's indeed a named element?

Reply:
Done.